yt_dlp/extractor/facebook.py

```diff
@@ -616,6 +603,32 @@ def parse_graphql_video(video):
                                 'url': playable_url,
                             })
                     extract_dash_manifest(video, formats)
+
+                    automatic_captions, subtitles = {}, {}
+                    is_video_broadcast = get_first([video], 'is_video_broadcast', expected_type=bool)
+                    captions = (traverse_obj(video,
+                                ('video_available_captions_locales', lambda _, v: 'captions_url' in v))
+                                or video.get('captions_url'))
+                    if url_or_none(captions):  # if video only had a 'captions_url'
+                        locale = self._html_search_meta(['og:locale', 'twitter:locale'], webpage, 'locale', default='en_US')
+                        auto_gen = (traverse_obj(video,
+                                    ('video_available_captions_locales', ..., 'localized_creation_method'), get_all=False)
+                                    or is_video_broadcast)
+                        (automatic_captions if auto_gen else subtitles)[locale] = [{'url': captions}]
+                    # or else video had 'video_available_captions_locales', a list of dicts
+                    for caption in traverse_obj(captions, (
+                        {lambda x: sorted(x, key=lambda c: c['locale'])}, lambda _, v: v['captions_url'])
+                    ):
```

Comment:
we don't have use the same var for the single `captions_url` and the `captions` list anymore
```suggestion
                    if captions_url := traverse_obj(video, ('captions_url', {url_or_none})):
                        locale = self._html_search_meta(
                            ['og:locale', 'twitter:locale'], webpage, 'locale', default='en_US')
                        auto_gen = is_video_broadcast or traverse_obj(video, (
                            'video_available_captions_locales', ..., 'localized_creation_method'))
                        (automatic_captions if auto_gen else subtitles)[locale] = [{'url': captions_url}]
                    for caption in traverse_obj(video, (
                        'video_available_captions_locales',
                        {lambda x: sorted(x, key=lambda c: c['locale'])},
                        lambda _, v: url_or_none(v['captions_url'])
                    )):
```

Reply:
For this subtitle part, the source dict feeded into `parse_graphql_video()` looks like this:

case 1:
only one subtitle (exception: FB Events give us incomplete value in `video_available_captions_locales` instead of `[]`)
```
{
    ...,
    'captions_url': 'https://...',
    'video_available_captions_locales': [],
    ...,
}
```
case 2: multiple subtitles:
```
{
    ...,
    'captions_url': 'https://...',
    'video_available_captions_locales': [
        {
            'localized_creation_method': None or 'Auto-generated',
            'captions_url': 'https://...',
            'locale': '?????',
            ...,
        }, {
            'localized_creation_method': None or 'Auto-generated',
            'captions_url': 'https://...',
            'locale': '?????',
        },
        ...,
    ],
    ...,
}
```
The value in `captions_url` (the outer one) is one of the url in `video_available_captions_locales` depended on locale (locale can be set via cookies). Therefore, ```captions = get_first(subs_data, 'video_available_captions_locales', 'captions_url')``` exists in the original codes in order to pick either `video_available_captions_locales` or `captions_url` (outer) with non-empty `video_available_captions_locales` in 1st priority (but FB Events make this line fail).

Regarding the suggestion, we will get this in case 2:
```
{
    'en_US': [
        {
            'url': 'https...'   # came from 'captions_url' (outer) when locale='en_US' ('en_US' from line 610)
        },
        {
            'url': 'https...',  # came from 'video_available_captions_locales' where 'locale'='en_US'
            'name': 'English'
        }
    ]
}
```
because it effectively picks both of the two.

Reply:
then let's do this
```suggestion
                    for caption in traverse_obj(video, (
                        'video_available_captions_locales',
                        {lambda x: sorted(x, key=lambda c: c['locale'])},
                        lambda _, v: url_or_none(v['captions_url'])
                    )):
                        lang = caption.get('localized_language', '')
                        subs = {
                            'url': caption['captions_url'],
                            'name': format_field(caption, 'localized_country', f'{lang} (%s)', default=lang),
                        }
                        if caption.get('localized_creation_method') or is_video_broadcast:
                            automatic_captions.setdefault(caption['locale'], []).append(subs)
                        else:
                            subtitles.setdefault(caption['locale'], []).append(subs)
                    else:
                        if captions_url := traverse_obj(video, ('captions_url', {url_or_none})):
                            locale = self._html_search_meta(
                                ['og:locale', 'twitter:locale'], webpage, 'locale', default='en_US')
                            (automatic_captions if is_video_broadcast else subtitles)[locale] = [{'url': captions_url}]
```